### ISSUE TITLE
initializing from VGG pretrained model

### ISSUE THREAD

**Issue by deepaksinghcv** [Sayan98/pytorch-segnet#3]
Hey,
In the model.py file, the  self.init_vgg_weigts() function has initialization of weights of 21,24,26th layer for the last 3 convolution layers, shouldn't it be 24,26,28th layer.
Kindly correct me if i'm wrong.
Thank you

**Comment by say4n** [comment]
Hey, @deepakksingh thanks for catching this!
It's fixed now. :)

**Comment by deepaksinghcv** [comment]
i have a question ,
for a random test image, how will i interpret the tensor, it's in the shape [batchsize,22,224,224], where 22 is the number of possible class labels and 224 is image width and height.
In the forward pass, I'm doing softmax after the decoder output.
what next?

Thank you.

**Comment by Sayan98** [comment]
In the task of semantic image segmentation, each pixel is assigned a class
label. So, doing softmax + argmax on the `n x n x c` tensor will help you
get the pixel-wise class labels.

**Comment by say4n** [comment]
In the task of [semantic image segmentation](http://www.cs.toronto.edu/~tingwuwang/semantic_segmentation.pdf), each pixel is assigned a class label. So, doing softmax + argmax on the `n x n x c` tensor will help you get the pixel-wise class labels.

**Comment by deepaksinghcv** [comment]
In your inference.py file at line 80:
predicted_mx = predicted_mx.argmax(axis=0)
shouldn't it be predicted_mx = predicted_mx[predicted_mx.argmax(axis=0)]
(or something similar to the above..so that we can fetch the maximum values)
so that we choose the max predictions from each channel.

**Comment by deepaksinghcv** [comment]
> In your inference.py file at line 80:
> predicted_mx = predicted_mx.argmax(axis=0)
> shouldn't it be predicted_mx = predicted_mx[predicted_mx.argmax(axis=0)]
> (or something similar to the above..so that we can fetch the maximum values)
> so that we choose the max predictions from each channel.

No worries, I solved that.
thank you
